processor/src/host/advice/mod.rs

```diff
@@ -41,9 +41,9 @@ type SimpleMerkleMap = BTreeMap<Word, StoreNode>;
 /// Advice data is store in-memory using [BTreeMap]s as its backing storage.
 #[derive(Debug, Clone, Default)]
 pub struct AdviceProvider {
-    stack: Vec<Felt>,
-    map: AdviceMap,
-    store: MerkleStore<SimpleMerkleMap>,
+    pub stack: Vec<Felt>,
+    pub map: AdviceMap,
+    pub store: MerkleStore<SimpleMerkleMap>,
 }
```

Comment:
Why do we need to make this public? We may refactor the advice provider such that it would require some additional logic on how the internal data could be accessed or modified. So, I'd rather keep the fields private.